### PR TITLE
fixes #4418. Since event firing is done via executor service, test rarel...

### LINKED ISSUE #4418: [TEST-FAILURE] MembershipListenerTest.whenMemberAdded_thenMemberAddedEvent
```
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:743)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:555)
	at org.junit.Assert.assertEquals(Assert.java:542)
	at com.hazelcast.client.MembershipListenerTest$1.run(MembershipListenerTest.java:65)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:251)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:267)
	at com.hazelcast.client.MembershipListenerTest.whenMemberAdded_thenMemberAddedEvent(MembershipListenerTest.java:62)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.maintenance-IbmJDK1.6/com.hazelcast$hazelcast-client/114/testReport/junit/com.hazelcast.client/MembershipListenerTest/whenMemberAdded_thenMemberAddedEvent/

### PR DESCRIPTION
**Pull request by sancar**
Since event firing is done via execeture service, test rarel.y gets the first node as added event too. Test is changed accordingly

### CODE REVIEW

File: hazelcast-client/src/test/java/com/hazelcast/client/MembershipListenerTest.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ public void whenMemberAdded_thenMemberAddedEvent() throws Exception {
         assertTrueEventually(new AssertTask() {
             @Override
             public void run() {
-                assertEquals(1, listener.events.size());
-                MembershipEvent event = listener.events.get(0);
+                assertTrue(1 <= listener.events.size());
```
Comment (pveentjer) [inline hazelcast-client/src/test/java/com/hazelcast/client/MembershipListenerTest.java]: The issue with assertTrue is that if it fails, the message often sucks. Can you improe the message to include the actual events.size? If we get an error, then we know how many there were.

File: hazelcast-client/src/test/java/com/hazelcast/client/MembershipListenerTest.java [hunk 2]
```diff
@@ -123,16 +123,16 @@ public void addNullListener_thenException() throws Exception {
     }
 
 
-    private class MemberShipEventLoger implements MembershipListener {
+    private class MemberShipEventLogger implements MembershipListener {
 
-        private List<MembershipEvent> events = new Vector<MembershipEvent>();
+        public LinkedBlockingDeque<MembershipEvent> events = new LinkedBlockingDeque<MembershipEvent>();
```
Comment (pveentjer) [inline hazelcast-client/src/test/java/com/hazelcast/client/MembershipListenerTest.java]: I'm fine with using a queue here, but you can also get the last element of a vector using vector.lastElement().

Comment (sancar) [inline hazelcast-client/src/test/java/com/hazelcast/client/MembershipListenerTest.java]: Good to know.

**Comment by sancar** [comment]
I opened an issue related to fail here https://github.com/hazelcast/hazelcast/issues/4561

**Comment by sancar** [comment]
verify

**Comment by eminn** [comment]
Looks good to merge.

**Pull request merged**